File: models/Gpt-3.5-turbo/cpp/code/problems/sorting_and_searching/Restaurant_Customers.cpp

#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;

    int arrivals[n], leavings[n];
    for (int i = 0; i < n; i++) {
        cin >> arrivals[i] >> leavings[i];
    }

    sort(arrivals, arrivals + n);
    sort(leavings, leavings + n);

    int current_count = 0, max_count = 0;
    int i = 0, j = 0;

    while (i < n && j < n) {
        if (arrivals[i] <= leavings[j]) {
            current_count++;
            i++;
        } else {
            current_count--;
            j++;
        }
        max_count = max(max_count, current_count);
    }

    cout << max_count << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/sorting_and_searching/Restaurant_Customers.py

n = int(input())
arrivals = []
leavings = []

for _ in range(n):
    a, b = map(int, input().split())
    arrivals.append(a)
    leavings.append(b)

arrivals.sort()
leavings.sort()

max_customers = 0
current_customers = 0
i = 0
j = 0

while i < n and j < n:
    if arrivals[i] < leavings[j]:
        current_customers += 1
        max_customers = max(max_customers, current_customers)
        i += 1
    else:
        current_customers -= 1
        j += 1

print(max_customers)
